staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch_test.go

```diff
@@ -654,6 +654,86 @@ func TestCustomStrategicMergePatch(t *testing.T) {
 	}
 }
 
+// TestStrategicMergeTreatNilCollectionAsEmpty tests the behavior where nil and empty collections in
+// the Go struct are treated equivalently.
+func TestStrategicMergeTreatNilCollectionAsEmpty(t *testing.T) {
+	for i, tc := range []struct {
+		original *MergeItem
+		modified *MergeItem
+		current  *MergeItem
+		want     []byte
+	}{{
+		// original->modified deletes the last key, but current has a custom key.
+		original: &MergeItem{SimpleMap: map[string]string{"original-key": "1"}},
+		modified: &MergeItem{SimpleMap: map[string]string{}},
+		current:  &MergeItem{SimpleMap: map[string]string{"original-key": "1", "custom-key": "2"}},
+		want:     []byte(`{"SimpleMap":{"original-key":null}}`),
+	}, {
+		// original->modified deletes the last key (with simulated omitempty behavior), but current has
+		// a custom key.
+		original: &MergeItem{SimpleMap: map[string]string{"original-key": "1"}},
+		modified: &MergeItem{}, // SimpleMap is nil (simulated omitempty).
+		current:  &MergeItem{SimpleMap: map[string]string{"original-key": "1", "custom-key": "2"}},
+		want:     []byte(`{"SimpleMap":{"original-key":null}}`),
```

Comment:
To be explicit: without my PR this fails with a merge conflict. But removing the last key often leads to this scenario since since `omitempty` is present on almost all API objects (e.g. https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta)